### ISSUE TITLE
Manually selected UDP Port gets displayed but not set up

### ISSUE THREAD

**Issue by kingsacrifice** [dannagle/PacketSender#130]
#### What OS?
- [ ] Windows
- [Mac OS Mojave v10.14.2] Mac
- [ ] Linux (Which distro?)

#### Description of issue

Packet Sender Version 5.8.6
Manually setting a UDP Port seems to work on the GUI (the port number gets displayed), but does not actually get used by the software.
Generating a random Port and using that port number works just fine.

Same behaviour confirmed independently on Windows 7 (Packet Sender v6.0.19)

**Comment by kingsacrifice** [comment]
Yes, all numbers match up, but only the randomly generated ones seem to work.

**Comment by dannagle** [comment]
Could that particular port be bad? It's possible a process crashed and never correctly released the port you are trying to bind. That could be why random ones work -- Packet Sender is requesting the OS for a free port.

**Comment by kingsacrifice** [comment]
I tried some other manually selected ports: numbers like 1083, 55555, and even previously working port numbers when they had been generated by the software. All of them didn't work for me.

Additionally I tried some other automatically generated port numbers. They did work.

**Comment by dannagle** [comment]
Checking for bad ports is difficult.  I usually just force-close the misbehaving app and/or reboot.

You can run **netstat** from the mac Terminal to see which ports are consumed.

The UDP of my Packet Sender on Mac is bound to port 5700.  I see that port listed here. 
```
Dans-MacBook-Pro:~ dannagle$ netstat -anp udp
Active Internet connections (including servers)
Proto Recv-Q Send-Q  Local Address          Foreign Address        (state)    
udp4       0      0  *.5700                 *.*                               
udp6       0      0  *.60118                *.*               

```

Maybe yours will show a bunch of stale ports.

**Comment by kingsacrifice** [comment]
Just looking at the udp part, there are 6 local addresses as listed by you *.number.
Additionally there are about 20 *.* ones and 8 *.mdns ones and a few others (*.some variable).
Couldn't find the non-working port numbers, though.

**Comment by kingsacrifice** [comment]
Changing to IPv4-only kind of helped. However when selecting a port manually, I need to send a message from my Mac to my project first before the other direction of communication works. As it turns out, it then does not matter which port number I select. 
However, when using a randomly generated port number in Packet Sender, I can directly transmit messages, without messaging from my Mac first.
So I suspect the manual selection does not work.
If you want to look at this more in-depth, I am happy to assist.
For my purposes this solution is sufficient so thanks a lot for your support!

I think that I found a small bug: When closing the Preferences window by clicking OK, every time the Preferences window is reopened, the selection is toggled between "IPv6-only" and "both IPv4 and IPv6" compared to it's previous state (the IPv4 selection remains if chosen). Closing the Preferences window by hitting cancel does not show this behaviour. 
From what I can tell, the selected state before closing is actually used and it seems to be a crossed assignment in the initialisation of radio buttons when opening the preferences window. But I am not exactly sure about which state gets actually used, so it might also be a crossed assignment in the read back from the preferences window.
